src/journal/views.py

```diff
@@ -1392,36 +1392,63 @@ def search(request):
     """
     articles = []
     search_term = None
-
-    if request.POST:
-        search_term = request.POST.get('search')
-        request.session['article_search'] = search_term
-        return redirect(reverse('search'))
-
-    if request.session.get('article_search'):
-        search_term = request.session.get('article_search')
-
-        article_search = submission_models.Article.objects.filter(
-            (Q(title__icontains=search_term) |
-             Q(subtitle__icontains=search_term)) &
-            Q(journal=request.journal)
-        )
-        article_search = [article for article in article_search]
-
-        author_search = search_term.split(' ')
-        from_author = submission_models.FrozenAuthor.objects.filter(
-            (Q(first_name__in=author_search) |
-             Q(last_name__in=author_search)) &
-            Q(article__journal=request.journal)
-        )
-
-        articles_from_author = [author.article for author in from_author]
-        articles = set(article_search + articles_from_author)
+    keyword = None
+    redir = False
+    sort = 'title'
+          
+    search_term, keyword, sort, form, redir = logic.handle_search_controls(request)
+
+    if redir:
+        return redir   
+
+    if search_term:
+        # checks titles, keywords and subtitles first, 
+        # then matches exact Author FN or LN
+        articles = submission_models.Article.objects.filter(
+                    (
+                        Q(title__icontains=search_term) |
+                        Q(keywords__word__icontains=search_term) |
+                        Q(subtitle__icontains=search_term)
+                    ) 
+                    |
+                    (
+                        Q(frozenauthor__first_name=search_term) |
+                        Q(frozenauthor__last_name=search_term)
```

Comment:
My apologies
So:
`frozenauthor__first/lastname__iexact=search_term` and `frozenauthor__fist/last_name__in=author_search)`

Reply:
if we keep a copy of `search_term` in `author_search` we can just `frozenauthor__fist/last_name__in=author_search`

Reply:
I think people expect case insensitivity when they are searching, also i think mobile users tend to not capitalize when doing these searches. But maybe that can be done with regex?

Reply:
Yeah, otherwise it will only be case insensitive for the CONTAINS filter and not for the IN filter.
Something like this horrible oneliner should work:
`author_search_regex = "^({})$".format("|".join(set(name  for name in set(chain(search_term.split(" "),(search_term,))))))`

The advantage of using the IN filter is that we could more easily index the table if this becomes a bit of a bottleneck but we will cross that bridge once we get to it.